src/tests/Common/helixpublishwitharcade.proj

```diff
@@ -429,61 +517,86 @@
       <MergedTestHelixCommand>$([System.IO.File]::ReadAllText('%(PayloadHelixCommandFile)'))</MergedTestHelixCommand>
       <PayloadGroup>$([System.IO.Path]::GetDirectoryName('%(MergedTestHelixCommand)').Replace('/', '-').Replace('\', '-'))</PayloadGroup>
       <PayloadZipFile>$(MergedPayloadsRootDirectory)%(PayloadGroup).zip</PayloadZipFile>
     </MergedPayloads>
+  </ItemGroup>
 
-    <MergedPayloads Update="@(MergedPayloads)" Condition="'$(TargetOS)' != 'windows'">
+  <ItemGroup Condition="'$(TargetsBrowser)' == 'true'">
+    <MergedPayloads Update="@(MergedPayloads)">
+      <PayloadHelixCommandFile>$(MergedPayloadsRootDirectory)%(FileName)%(Extension)\HelixCommand.txt</PayloadHelixCommandFile>
+      <MergedTestHelixCommand>$([System.IO.File]::ReadAllText('%(PayloadHelixCommandFile)'))</MergedTestHelixCommand>
+      <PayloadGroup>%(FileName)%(Extension)</PayloadGroup>
+      <PayloadZipFile>$(MergedPayloadsRootDirectory)%(PayloadGroup).zip</PayloadZipFile>
+    </MergedPayloads>
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetsMobile)' != 'true' or '$(TargetsBrowser)' == 'true'">
+    <MergedPayloads Update="@(MergedPayloads)" Condition="'$(TestWrapperTargetsWindows)' != 'true'">
       <MergedTestHelixCommand>chmod +x %(MergedPayloads.MergedTestHelixCommand)%0A%(MergedPayloads.MergedTestHelixCommand)</MergedTestHelixCommand>
     </MergedPayloads>
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetsMobile)' == 'true' and '$(TargetsBrowser)' != 'true'">
+    <MergedPayloads Update="@(MergedPayloads)">
+      <PayloadGroup>%(FileName)%(Extension)</PayloadGroup>
+      <ApkPath>$(MergedPayloadsRootDirectory)%(FileName)%(Extension)/%(FileName)%(Extension).apk</ApkPath>
+      <AppBundlePath>$(MergedPayloadsRootDirectory)%(FileName)%(Extension)/%(FileName)%(Extension).app</AppBundlePath>
+    </MergedPayloads>
+  </ItemGroup>
 
-    <HelixWorkItem Include="@(LegacyPayloads->Metadata('PayloadGroup'))" Condition=" '$(IsRunningOnMobileTargets)' == 'false' ">
+  <ItemGroup>
+    <HelixWorkItem Include="@(LegacyPayloads->Metadata('PayloadGroup'))" Condition=" '$(TargetsMobile)' != 'true' or '$(TargetsBrowser)' == 'true' ">
       <PayloadDirectory>%(PayloadDirectory)</PayloadDirectory>
       <Command>dotnet $(XUnitRunnerDll) %(XUnitWrapperDlls) $(XUnitRunnerArgs)</Command>
       <Command Condition=" '%(TestGroup)' != '' ">dotnet $(XUnitRunnerDll) %(XUnitWrapperDlls) $(XUnitRunnerArgs) -trait TestGroup=%(TestGroup)</Command>
       <Timeout Condition=" '$(TimeoutPerTestCollectionInMinutes)' != '' ">$([System.TimeSpan]::FromMinutes($(TimeoutPerTestCollectionInMinutes)))</Timeout>
     </HelixWorkItem>
 
-    <HelixWorkItem Include="@(MergedPayloads->Metadata('PayloadGroup'))" Condition=" '$(IsRunningOnMobileTargets)' == 'false' ">
+    <HelixWorkItem Include="@(MergedPayloads->Metadata('PayloadGroup'))" Condition=" '$(TargetsMobile)' != 'true' or '$(TargetsBrowser)' == 'true' ">
```

Comment:
Nit: I am seeing at least four places that query this particular combination of conditions - ("browser or not mobile" and its negation "non-browser on mobile"), would it make sense to synthesize a new property for this purpose?

Reply:
I can't think of a good name for a consolidated property, so I'd prefer to keep using our existing properties as-is, at least for now.

Reply:
Is what we are really testing here whether or not the target uses a custom host (i.e. something other than corerun?) We could call it that; "TargetUsesCustomHost"?

Reply:
I've made this condition `TargetHasHelixXHarnessSdkSupport` as that's what best describes most of the conditions where this condition or its inverse is used.